contracts/TokenSale.sol

```diff
@@ -214,11 +214,18 @@ contract TokenSale is OpsManaged, Pausable, TokenSaleConfig { // Pausable is als
 
 
     function hasSaleEnded() private view returns (bool) {
-        if (totalTokensSold >= TOKENS_SALE || currentTime() >= endTime || finalized) {
+        // if sold out or finalized, sale has ended
+        if (totalTokensSold >= TOKENS_SALE || finalized) {
             return true;
+        // else if sale is not paused (pausedTime = 0) 
+        // and endtime has past, then sale has ended
+        } else if (pausedTime == 0 && currentTime() >= endTime) {
+            return true;
+        // otherwise it not past and not paused; or paused
+        // and as such not ended
+        } else {
+            return false;
```

Comment:
Now that `hasSaleEnded` will return false if the sale is paused and we have past the original `endTime`, `updateWhitelist` should probably stet `require(!hasSaleEnded())`, because otherwise the whitelist can be updated after the sale ends but before it is finalized.

Thoughts?

Reply:
Great point; will make said changes.